main.cpp

```diff
@@ -7,12 +7,32 @@
 #include "toz3Options.h"
 
 
+class Pruner : public Transform{
+    public: 
+        Pruner(){
+            setName("Pruner");
+        }
+
+        IR::Node* preorder(IR::Statement *s);
+
+};
+
+
+IR::Node* Pruner::preorder(IR::Statement *s){
+    if(s->node_type_name()== "AssignmentStatement"){
+
+        return nullptr;
+}    else
+        return s;
+
+}
+
```

Comment:
I would add this in a separate file. The visitor code will become fairly large soon.